pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -14,11 +14,22 @@ rec {
   mkDerivation =
     { name ? ""
 
-    , nativeBuildInputs ? []
-    , buildInputs ? []
+    # TODO(@Ericson2314): Stop using legacy dep attribute names
+    #                               host offset -> target offset
+    , depsBuildBuild               ? []    # -1 -> -1
+    , depsBuildBuildPropagated     ? []    # -1 -> -1
+    , nativeBuildInputs            ? []    # -1 -> 0  N.B. Legacy name
+    , propagatedNativeBuildInputs  ? []    # -1 -> 0  N.B. Legacy name
+    , __depsBuildTarget            ? []    # -1 -> 1
+    , __depsBuildTargetPropagated  ? []    # -1 -> 1
```

Comment:
It would be great to see documentation for all new variables (and as bonus even for the legacy variables) at some point.